service/service.go

```diff
@@ -69,9 +70,10 @@ type Service struct {
 	telemetrySettings    component.TelemetrySettings
 	host                 *serviceHost
 	telemetryInitializer *telemetryInitializer
+	collectorConf        *confmap.Conf
 }
 
-func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
+func New(ctx context.Context, set Settings, cfg Config, collectorConf *confmap.Conf) (*Service, error) {
```

Comment:
Please follow the conventions we have, conf should be part of settings. Also this is a breaking change.